qiskit_experiments/library/tomography/basis/local_basis.py

```diff
@@ -156,31 +123,118 @@ def circuit(
         return _tensor_product_circuit(self._instructions, index, self._name)
 
     def matrix(self, index: Sequence[int], qubits: Optional[Sequence[int]] = None):
+        # Convert args to hashable tuples
         if qubits is None:
             qubits = tuple(range(len(index)))
+        else:
+            qubits = tuple(qubits)
+        index = tuple(index)
+
         try:
+            # Look for custom POVM for specified qubits
+            state = self._generate_qubits_state(index, qubits)
+            if state is not None:
+                return state.data
+
             mat = np.eye(1)
-            for i, qubit in zip(index, qubits):
-                states = self._qubit_states.get(qubit, self._default_states)
-                mat = np.kron(states[i], mat)
+            for idx, qubit in zip(index, qubits):
+                qubit_state = self._generate_qubits_state((idx,), (qubit,))
+                mat = np.kron(qubit_state, mat)
             return mat
+
         except TypeError as ex:
             # This occurs if basis is constructed with qubit_states
             # kwarg but no default_states or instructions and is called for
             # a qubit not in the specified kwargs.
             raise ValueError(f"Invalid qubits for basis {self.name}") from ex
 
+    @cache_method()
+    def _generate_qubits_state(self, index: Tuple[int, ...], qubits: Tuple[int, ...]):
+        """LRU cached function for returning POVMS"""
+        num_qubits = len(qubits)
+
+        # Check for N-qubit states
+        if qubits in self._qubit_states:
+            # Get states for specified qubits
+            # TODO: In the future we could add support for different orderings
+            #       of qubits by permuting the returned POVMS
```

Comment:
Yeah, though I was initially thinking even simpler: that if you initialized with correlated povms on a set of qubits (a, b, c), you could handle any permutation of those qubits (b, c, a) etc. Then one could generalize the automatic povm calculation in the pauli basis classes to use a correlated readout mitigator instead of just the local readout mitigator.